http/webserver.go

```diff
@@ -187,7 +191,8 @@ func HandleStatic(h *config.WebStatic, w http.ResponseWriter, r *http.Request) (
 		} else if os.IsPermission(err) {
 			// If we tried opening a directory, we may not have permission to read it, but
 			// still access files inside it (execute bit), such as index.html. So try to serve it.
-			index, err := os.Open(filepath.Join(fspath, "index.html"))
+			var index *os.File
+			index, err = os.Open(filepath.Join(fspath, "index.html"))
```

Comment:
was this change meant as a way to prevent shadowing the "err" variable?
this one and/or the changes immediately below are causing test failures. the handling of these requests is somewhat tricky.

btw, i'm adding some comments. no expecting you to make changes to this PR at the moment. also see the issue for more info.